mne_sandbox/connectivity/viz.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+def plot_phase_locked_amplitude(epochs, freqs_phase, freqs_amp,
+                                ix_ph, ix_amp, mask_times=None,
+                                normalize=True,
+                                tmin=-.5, tmax=.5, return_data=False,
+                                amp_kwargs=None, ph_kwargs=None):
+    """Make a phase-locked amplitude plot.
+
+    Parameters
+    ----------
+    epochs : mne.Epochs
+        The epochs to be used in phase locking computation
+    freqs_phase : np.array
+        The frequencies to use in phase calculation. The phase of each
+        frequency will be averaged together.
+    freqs_amp : np.array
+        The frequencies to use in amplitude calculation.
+    ix_ph : int
+        The index of the signal to be used for phase calculation
+    ix_amp : int
+        The index of the signal to be used for amplitude calculation
+    normalize : bool
+        Whether amplitudes are normalized before averaging together. Helps
+        if some frequencies have a larger mean amplitude than others.
+    tmin : float
+        The time to include before each phase peak
+    tmax : float
+        The time to include after each phase peak
+    return_data : bool
+        If True, the amplitude/frequency data will be returned
+    amp_kwargs : dict
+        kwargs to be passed to pcolormesh for amplitudes
+    ph_kwargs : dict
+        kwargs to be passed to the line plot for phase
+
+    Returns
+    -------
+    axs : array of matplotlib axes
+        The axes used for plotting.
+    """
+    from .cfc import phase_locked_amplitude
+    from sklearn.preprocessing import scale
+    amp_kwargs = dict() if amp_kwargs is None else amp_kwargs
+    ph_kwargs = dict() if ph_kwargs is None else ph_kwargs
+
+    data_amp, data_phase, times = phase_locked_amplitude(
```

Comment:
so basically the `phase_amplitude_coupling` function is what calculates pairwise PAC between indices of electrodes. It always follows the structure of pacpy, and only returns a single PAC value in the same way that, e.g., spectral connectivity does.

`phase_locked_amplitude` returns actual data, but no PAC metric. It just calculates the amplitudes and phases, stacks them according to phase peaks, and then averages across the stack to return the mean phase/amplitude for the frequencies chosen. This lets you visualize it easily.

`phase_binned_amplitude` just bins the amplitude for high frequencies according to the phase of the low frequency, and then returns this data so you can plot a circular histogram. It also doesn't return any actual PAC value.

Most of the functions that `phase_amplitude_coupling` uses are under-the-hood functions that do filtering more efficiently assuming that you're passing multiple indices for connectivity pairs.